api/bind.go

```diff
@@ -92,7 +98,14 @@ func Bind(r *http.Request, dest interface{}) error {
 	}
 
 	if r.Header.Get("Content-Type") == "application/json" {
-		decoder := json.NewDecoder(r.Body)
+		reader := readerPool.Get().(*bufio.Reader)
+		defer func() {
+			reader.Reset(nil)
```

Comment:
https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/bufio/bufio.go;l=71

i think this beats the purpose of the pool because it looks like Reset(nil) allocates a new buffer

is it maybe impossible to reuse a bufio.Reader safely?